### PR TITLE
Add Services Parameter

### PR DESCRIPTION
**Pull request by devinlyons**
The `Services` parameter allows the plugin optionally to query the health of specific services.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -88,32 +93,17 @@ func executeCheck(event *types.Event) (int, error) {
...
-	defer resp.Body.Close()
 	if err != nil {
-		return sensu.CheckStateCritical, err
-	}
-
-	var sResp ServiceResponse
-
-	if err := json.NewDecoder(resp.Body).Decode(&sResp); err != nil {
-		return sensu.CheckStateCritical, fmt.Errorf("Failed to decode service response: %v", err)
+		return sensu.CheckStateCritical, fmt.Errorf("could not retrive service health: %v", err)
```
Comment (wduncanfraser) [inline main.go]: retrieve

**Review by wduncanfraser — CHANGES_REQUESTED** [review]
This is moving in the right direction.

The one thing we want to do is make sure we are always using the health check endpoints, due to real time data concerns. The census and service summary endpoints are based on gossip data and don't always reflect the point in time health check status like the specific health check endpoints do.

http://localhost:9631/services/postgresql11/default/health for example, vs parsing the output of http://localhost:9631/services/postgresql11/default for the health check.

We probably want the workflow to be:
If list is provided, query each individual health check endpoint (like you are currently doing for the individual service endoints in checkServices).
If list is not provided, query the services endpoint, build a list, and shove it into checkServices.

File: main.go [hunk 2]
```diff
@@ -153,3 +150,75 @@ func executeCheck(event *types.Event) (int, error) {
...
+	return result, nil
+}
+
+func getResponse(client *http.Client, req *http.Request) (*http.Response, error) {
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
```
Comment (wduncanfraser) [inline main.go:219]: In golang, defer delays execution of a function call until the enclosing function returns. In this case, we will close the body immediately due to this function returning and then try to access it in the calling function, which will result in an error.

We would need to move the body close out of this function to the call site, which limits the usefulness of the abstraction. We probably just want to drop this function and replace calls to getResponse with the direct Client.Do/Body.Close interaction.

**Review by wduncanfraser — CHANGES_REQUESTED** [review]
Definitely getting there! Tried running it locally, and ran into issues with the http client execution that I left a comment on.

Once that is resolved, I would recommend running the check locally against your local supervisor with `go run main.go`, testing the possible permutations of the check to see if we get back useful messaging in the stdout: 

- Allow it to check all services automatically
- Explicitly check for services you have running
- Explicitly check for services you don't have running
- Stop a local running service and see what the check reports in both all service and explicit service checks

We definitely want to make sure that if a service is down, we get back messaging that "Service X is in state Y (or that we failed to parse the response)", and that the check continues processing the rest of the services, so we know if multiple are down.

File: main.go (hunk 2 above)
Comment (devinlyons) [reply main.go:219]: Interesting. I can see how `defer` might be useful. I will eliminate that utility function. It made more sense to have it before the second commit.

**Comment by devinlyons** [comment]
I ran the plugin against my local environment and confirmed the checks worked.

**Review by wduncanfraser — APPROVED** [review]
Awesome, this looks good to me now.

**Pull request merged**